extensions/amp-date-countdown/0.1/test/test-amp-date-countdown.js

```diff
@@ -0,0 +1,235 @@
+    expect(itemElement.textContent)
+        .to.equal('02:00:00:00');
+  });
+
+  it('should display timeleft in the format ' +
+    '{h} {hours} and {m} {minutes} and {s} {seconds}', () => {
+    element.build();
+    const timeObj =
+      Object.assign(
+          impl.getYDHMSFromMs_((endDate - twoDaysBeforeEndDate) - 1000),
+          impl.getLocaleWord_('en')); //English
+    const itemElement = win.document.createElement('div');
+    itemElement.textContent = timeObj.h + ' ' + timeObj.hours + ' and ' +
+      timeObj.m + ' ' + timeObj.minutes + ' and ' +
+      timeObj.s + ' ' + timeObj.seconds;
+    expect(itemElement.textContent)
+        .to.equal('23 Hours and 59 Minutes and 59 Seconds');
+    //1 day 23 Hours and 59 minutes and 59 seconds
+  });
+
+  it('should display timeleft in the format ' +
+    '{d} {days} {h}:{mm}', () => {
+    element.build();
+    const timeObj =
+      Object.assign(
+          impl.getYDHMSFromMs_((endDate - twoDaysBeforeEndDate) - 1000),
+          impl.getLocaleWord_('en')); //English
+    const itemElement = win.document.createElement('div');
+    itemElement.textContent = timeObj.d + ' ' + timeObj.days + ' ' +
+      timeObj.h + ':' + timeObj.mm;
+    expect(itemElement.textContent)
+        .to.equal('1 Days 23:59');
+  });
+
+  it('should calculate the timeleft after added offset-seconds', () => {
+    element.build();
+    const timeObj =
+      Object.assign(
+          impl.getYDHMSFromMs_(endDate - twoDaysBeforeEndDate
+            + (24 * 60 * 60 * 1000)), // hours * minutes * seconds * ms
+          impl.getLocaleWord_('en')); // English
+    const itemElement = win.document.createElement('div');
+    itemElement.textContent = timeObj.d + ' ' + timeObj.days + ' ' +
+      timeObj.h + ' ' + timeObj.hours + ' ' +
+      timeObj.m + ' ' + timeObj.minutes + ' ' +
+      timeObj.s + ' ' + timeObj.seconds ;
+    expect(itemElement.textContent)
+        .to.equal('3 Days 0 Hours 0 Minutes 0 Seconds');
+  });
+
+  it('should calculate the timeleft after substracted offset-seconds', () => {
+    element.build();
+    const timeObj =
+      Object.assign(
+          impl.getYDHMSFromMs_(endDate - twoDaysBeforeEndDate
+            + (-1 * 24 * 60 * 60 * 1000)), // hours * minutes * seconds * ms
+          impl.getLocaleWord_('en')); // English
+    const itemElement = win.document.createElement('div');
+    itemElement.textContent = timeObj.d + ' ' + timeObj.days + ' ' +
+      timeObj.h + ' ' + timeObj.hours + ' ' +
+      timeObj.m + ' ' + timeObj.minutes + ' ' +
+      timeObj.s + ' ' + timeObj.seconds ;
+    expect(itemElement.textContent)
+        .to.equal('1 Days 0 Hours 0 Minutes 0 Seconds');
+  });
+
+  // it('should calculate timeleft and have biggest-unit up to hours', () => {
+  //   element.setAttribute('biggest-unit', 'HOURS');
+  //   element.build();
+  //   let timeObj;
+  //   let itemElement;
+  //   Services.viewerForDoc(win.document).whenFirstVisible().then(() => {
```

Comment:
because this test is `async` you need to add a `return` here and return the `promise` so that the test runner knows to wait.

Reply:
looks like you are stubbing the `whenFirstVisible` above in test setup and keeping a reference under `viewer` variable so
```
return viewer.whenFirstVisible().then() {
});
whenFirstVisiblePromiseResolve(); 
```

should work.

Reply:
nope, still not working as expected

Reply:
feel free to skip for now to get the PR in, we can fix later.

Reply:
Okay, Already connected them out , and the rest of the test scripts are passed !

Reply:
@leonalicious there seems to be a conflict in `gulp.js` after that's resolved and build is green, we can merge. Thanks for all the work

Reply:
here you go, should be resolved by now~

Reply:
a new unit test failure (32:7  error  'whenFirstVisiblePromiseResolve' is assigned a value but never used. Allowed unused vars must match /AmpElement|Def|Interface$/  no-unused-vars
)

(we are constantly improving our lint/jsdoc rules so that is why these issues keep coming up, so as you rebase and try to merge PR, there might be new rules to adhere to that did not exist before. Apologies for the inconvenience this causes but needed for code quality improvements)

Reply:
i do run `sudo gulp pr-check` and notice there are too many errors which flood my entire command screen and i couldn't not pin-point and check on `amp-date-countdown` properly. but i guess all linter check should be achieved by now~